python_modules/libraries/dagster-dg/dagster_dg/cli/plus/deploy.py

```diff
@@ -0,0 +1,290 @@
+    if not organization:
+        raise click.UsageError(
+            "Organization not specified. To specify an organization, use the --organization option "
+            "or run `dg plus login`."
+        )
+    return organization
+
+
+def _get_deployment(input_deployment: Optional[str], plus_config: DagsterPlusCliConfig) -> str:
+    deployment = input_deployment or plus_config.default_deployment
+    if not deployment:
+        raise click.UsageError(
+            "Deployment not specified. To specify a deployment, use the --deployment option "
+            "or run `dg plus login`."
+        )
+    return deployment
+
+
+org_and_deploy_group = make_option_group(
+    {
+        not_none(option.name): option
+        for option in [
+            click.Option(
+                ["--organization"],
+                "organization",
+                help="Dagster+ organization to which to deploy. If not set, defaults to the value set by `dg plus login`.",
+                envvar="DAGSTER_PLUS_ORGANIZATION",
+            ),
+            click.Option(
+                ["--deployment"],
+                "deployment",
+                help="Name of the Dagster+ deployment to which to deploy (or use as the base deployment if deploying to a branch deployment). If not set, defaults to the value set by `dg plus login`.",
+                envvar="DAGSTER_PLUS_DEPLOYMENT",
+            ),
+        ]
+    }
+)
+
+
+@click.command(name="start-deploy-session", cls=DgClickCommand)
+@org_and_deploy_group
+@click.option(
+    "--deployment-type",
+    "deployment_type_str",
+    type=click.Choice([deployment_type.value for deployment_type in DgPlusDeploymentType]),
+    help="Whether to deploy to a full deployment or a branch deployment. If unset, will attempt to infer from the current git branch.",
+)
+@click.option(
+    "-y", "--yes", "skip_confirmation_prompt", is_flag=True, help="Skip confirmation prompts."
+)
+@dg_global_options
+@cli_telemetry_wrapper
+def start_deploy_session_command(
+    organization: Optional[str],
+    deployment: Optional[str],
+    deployment_type_str: Optional[str],
+    skip_confirmation_prompt: Optional[bool],
+    **global_options: object,
+) -> None:
+    """Start a new deploy session. Determines which code locations will be deployed and what
+    deployment is being targeted (creating a new branch deployment if needed), and initializes a
+    folder on the filesystem where state about the deploy session will be stored.
+    """
+    cli_config = normalize_cli_config(global_options, click.get_current_context())
+    plus_config = DagsterPlusCliConfig.get()
+    organization = _get_organization(organization, plus_config)
+    deployment = _get_deployment(deployment, plus_config)
+
+    # TODO This command should work in a workspace context too and apply to multiple projects
+    dg_context = DgContext.for_project_environment(Path.cwd(), cli_config)
+
+    statedir = _get_statedir()
```

Comment:
Is removing the ability to override by the CLI option intentional?

Reply:
yes, this is intentional. That seems rare enough to me that doing it via env var makes sense to me vs polluting the help command.